### PR TITLE
Documentation improvement for TaylorDiagram class and set_titles_and_labels

### LINKED ISSUE #176: Taylor Diagram class is not very robust
Things that have been giving me trouble when working on the Taylor Diagram example:

- [ ] need `taylor.add_xgrid(np.array([0.6, 0.9]))` to enforce X-Y ratio. This shouldn't be necessary. User should have the right ratio without needing to specify diagonal lines from origin to correlation
- [x]  `dia.add_model_set(stddev=[0.6], corrcoef=[0.24])` says it accepts floats, but I need to put the values in a list for it to not raise an error.
- [x] Should `add_xgrid()` and `add_ygrid()`be renamed to `add_corr_grid()` and `add_stddev_grid()` since the x and y nature of this plot isn't that intuitive?
- [ ] How to treat negative correlations? Add optional second quadrant to the left.
- [x] Changing figure size messes with legend locations - add note that best at 12,12
- [ ]  Add arrow annotation support to draw directional association between same variable of different models.

### PR DESCRIPTION
**Pull request by jukent**
## PR Summary

Closes #176 and #212

## PR Checklist

**General**
- [x] Make an issue if one doesn't already exist
- [x] Link the issue this PR resolves by adding `closes #XXX` to the PR description where XXX is the number of the issue.
- [x] Add a brief summary of changes to `docs/release-notes.rst` in a relevant section for the next unreleased release. Possible sections include: Documentation, New Features, Bug Fixes, Internal Changes, Breaking Changes/Deprecated
- [x] Add appropriate labels to this PR
- [x] Make your changes in a forked repository rather than directly in this repo
- [x] Open this PR as a draft if it is not ready for review
- [x] Convert this PR from a draft to a full PR before requesting reviewers
- [x] Passes `precommit`. To set up on your local, run `pre-commit install` from the top level of the repository. To manually run pre-commits, use `pre-commit run --all-files` and re-add any changed files before committing again and pushing.

**Documentation**
- [x] Docstrings have been added to all new functions
- [x] Docstrings have been updated with any function changes
- [x] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [x] User facing functions have been added to `docs/user_api/index.rst` under their module

**Examples**
- [x] Any new notebook examples added to `docs/examples/` folder
- [x] Clear all notebook cells
- [x] New notebook files added to `docs/examples.rst` toctree
- [x] New notebook files added to new entry in `docs/gallery.yml` with appropriate thumbnail photo in `docs/_static/thumbnails/`

### CODE REVIEW

File: src/geocat/viz/taylor.py [hunk 1]
```diff
@@ -423,14 +428,29 @@ def add_xgrid(self,
...
-        """Add gridlines (radii) to the Y axis (standard deviation) specified
-        by array *arr*
+        """Add gridlines to the correlation axis specified by array *arr*.
+
+        This function will be deprecated in favor of `add_corr_grid()`
+        """
+
+        return self.add_corr_grid(arr, color, linestyle, linewidth, **kwargs)
+
+    def add_stdev_grid(self,
```
Comment (kafitzgerald) [inline src/geocat/viz/taylor.py:444]: Pandas, numpy, and scipy all use "std" as the abbreviation for standard deviation.  Thoughts on using that instead for consistency with the broader ecosystem?

File: src/geocat/viz/taylor.py [hunk 2]
```diff
@@ -423,14 +428,29 @@ def add_xgrid(self,
...
+                       arr: typing.Union[xr.DataArray, np.ndarray, list, float],
+                       color: str = 'lightgray',
+                       linestyle=(0, (9, 5)),
+                       linewidth: int = 1,
+                       **kwargs):
+        """Add gridlines (radii) to the stard deviation axis specified by
```
Comment (kafitzgerald) [inline src/geocat/viz/taylor.py:450]:
```suggestion
        """Add radial gridlines to the standard deviation axis specified by
```

File: src/geocat/viz/util.py [hunk 3]
```diff
@@ -556,6 +556,8 @@ def set_titles_and_labels(ax: typing.Union[matplotlib.axes.Axes,
     >>> |                   Axes                    |
     >>> |                                           |
 
+    Be aware that the `suptitle` functionality does not automaticallly render well for Cartopy plots. Recommended fixes are to iterate on your figure size until it does not extend as far past your plot OR to use string manipulation to add enough spaces to center your left or right title keyword argument.
```
Comment (kafitzgerald) [inline src/geocat/viz/util.py:559]: Should we add an admonition to the docs for this?

I'm not sure how good of practice it is, but conceivably we could also reference that issue for some longer form suggestions.

File: src/geocat/viz/taylor.py (hunk 2 above)
Comment (kafitzgerald) [reply src/geocat/viz/taylor.py:450]: I think this is more consistent w/ matplotlib terminology.

File: src/geocat/viz/taylor.py [hunk 4]
```diff
@@ -44,6 +44,11 @@ class TaylorDiagram(object):
     stdLevel : list
         Optional list of tick locations for stddev axis
 
+    Notes
+    -----
+    Rendering of Taylor Diagrams looks best with a figure that is at least 10x10. If you  make your figure size too small,
```
[the remark this answers is not in the document]
Comment (kafitzgerald) [reply src/geocat/viz/taylor.py:49]: Also wondering about some sort of admonition here for the docs.

**Review by kafitzgerald — CHANGES_REQUESTED** [review]
Agree this makes more sense.

A few typos and suggestions mentioned in the comments, but mostly wanted to raise a few questions about terminology (while we're making painful changes).

I wonder if we want to add some sort of deprecation warnings as well.  Thoughts?

File: src/geocat/viz/taylor.py (hunk 1 above)
Comment (jukent) [reply src/geocat/viz/taylor.py:444]: Okay that's good to know, We should use it too then.

File: src/geocat/viz/util.py (hunk 3 above)
Comment (jukent) [reply src/geocat/viz/util.py:559]: Looking at it again, I think your original reply that he needed to use an appropriate figure size is the best solution. He had set his height to 11 inches, but if you use 4 it works just fine. That is common practice when dealing with Cartopy's vertical rendering spreading out to take up all of the space given to it by the figure. This now seems common enough that I don't think we need to link to the issue or have an extra annotation.

Here's another example: https://stackoverflow.com/questions/73826115/cartopy-too-much-vertical-space-between-subplots-even-with-tight-layout

File: src/geocat/viz/taylor.py (hunk 1 above)
Comment (jukent) [reply src/geocat/viz/taylor.py:444]: Will be done in my upcoming push

**Review by kafitzgerald — CHANGES_REQUESTED** [review]
I might have spaced this before (apologies if so), but we should probably add either deprecation warnings or pending deprecation warnings and something in a deprecations section of the release notes about the renamed methods.

**Comment by jukent** [comment]
> I might have spaced this before (apologies if so), but we should probably add either deprecation warnings or pending deprecation warnings and something in a deprecations section of the release notes about the renamed methods.
> 
> Otherwise, looks good.

Thanks @kafitzgerald just added them!

File: docs/release-notes.rst [hunk 5]
```diff
@@ -5,6 +5,15 @@
...
 =============
 
+v2024.04.1 (April 1, 2024)
+------------------------------
+This release ...
+
+Documentation
+^^^^^^^^^^^^^
+* Document known issue with `suptitle` argument of  ``set_titles_and_labels`` and Cartopy plots by `Julia Kent`_ in (:pr:`219`)
+* Add alias to ``Taylor_Diagram`` to have more meaningful keyword arguments by `Julia Kent`_ in (:pr:`219`)
```
Comment (kafitzgerald) [inline docs/release-notes.rst:15]:
```suggestion

Deprecations
^^^^^^^^^^
* Pending deprecation warnings added for ``TaylorDiagram`` methods ``add_xgrid`` and ``add_ygrid`` which are changing to the new ``add_corr_grid`` and ``add_std_grid`` by `Julia Kent`_ in (:pr:`219`)

```

Comment (kafitzgerald) [reply docs/release-notes.rst:15]: Not committed to this wording / formatting, but I think we should probably have something in here about the deprecations.

**Review by kafitzgerald — CHANGES_REQUESTED** [review]

File: docs/release-notes.rst (hunk 5 above)
Comment (jukent) [reply docs/release-notes.rst:15]: Oh good point. Slipped my mind trying to make the change quickly. I'll get to this on Monday.

**Review by kafitzgerald — APPROVED** [review]

**Pull request merged**